h264/read.go

```diff
@@ -301,3 +304,20 @@ func (b *BitReader) StreamPosition() (int, int, int) {
 func (b *BitReader) LogStreamPosition() {
 	logger.Printf("debug: %d byte stream @ byte %d bit %d\n", len(b.bytes), b.byteOffset, b.bitOffset)
 }
+
+type ue struct {
+	loc  *int
```

Comment:
Why use a pointer here?

Reply:
so I can give the fields as args and set them.

Reply:
That's not a nice way to do things.

Reply:
Then what's a nice way to do things ? Please give alternatives with criticisms.

Reply:
If you are mutating a value, return the mutated value.

`func mutator(v ue) ue`

Reply:
see comment sps.go:534